### ISSUE TITLE
Bad parameterized type serialization

### ISSUE THREAD

**Issue by sheinbergon** [OpenHFT/Chronicle-Queue#555]
Given this test code ( annotations/val are provided by lombok)

```
public class ChronicleQueueTest {
    
    @Data
    @Accessors(fluent = true, chain = true)
    public static class ParamterizedData<V> implements Marshallable {
        private V value;
    }


    @Data
    @Accessors(fluent = true, chain = true)
    public static class UnparamterizedData implements Marshallable {
        private BigInteger value;
    }

    @Data
    @Accessors(fluent = true, chain = true)
    public static class TestMessage implements Marshallable {
        ParamterizedData<BigInteger> parameterized;
        UnparamterizedData unparameterized;
    }

    public interface Processor {
        void method(TestMessage message);
    }

    public static void main(String[] args) throws IOException {
        val queue = ChronicleQueue.singleBuilder("/tmp/test")
                .rollCycle(RollCycles.DAILY)
                .build();

        val integer = BigInteger.valueOf(2142L);

        try (queue) {
            Processor processor = (var message) -> {
                System.out.println(message);
                System.out.println(message.unparameterized);
                val int1 = message.unparameterized.value;
                System.out.println(int1);
                System.out.println(message.parameterized);
                val int2 = message.parameterized.value;
                System.out.println(int2);
            };
            val writer = queue.methodWriter(Processor.class);
            val reader = queue.createTailer().methodReader(processor);
            val message = new TestMessage()
                    .unparameterized(new UnparamterizedData().value(integer))
                    .parameterized(new ParamterizedData<BigInteger>().value(integer));
            writer.method(message);
            reader.readOne();
        } finally {
            FileUtils.deleteDirectory(new File("/tmp/test"));
        }
    }
}
```

You will get a class cast exception on the line `val int2 = message.parameterized.value;`
due to the fact chronicle queue serializes paramterized types as strings, regardless of their actual type.
While this might be ok from some programming aspect, it's a really a big fault in terms of de/serialization.

Any help would be advised.

**Comment by sheinbergon** [comment]
Ok, so I resolved this by overriding `writeMarshallable` and `readMarshallable` and made sure to save the generic parameter type as `typeLiteral` during serialization and use accordingly during deserialization.
works like a charm.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Queue-5.17.2](https://github.com/OpenHFT/Chronicle-Queue/releases/chronicle-queue-5.17.2), [BOM-2.17.92](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.92)
